body.tex

```diff
@@ -0,0 +1,440 @@
+
+
+\subsection{Versioning and Naming} \label{sec:versioning}
+DM will  switch to Semantic Versioning \footnote{\url{https://semver.org/spec/v2.0.0.html}},  this will help improve dependency management, and make  managing patch releases more comprehensible.
+The versioning schema will have 3 digits separated by a "." . For example :
+\\
+$16.0.0$
+
+Generalizing, the versioning scheme is the following:
+
+$M.n.p$
+
+where
+
+\begin{itemize}
+\item \textbf{M} is the major version identifier,
+\item \textbf{n} is the minor version identifier,
+\item \textbf{p} is the patch level identifier
+\end{itemize}
+
+The repository at this stage would be tagged $16.0.0$ .
+One should note in this scheme there are no padding zeros thus:
+\\
+$16.1.0   \neq 16.10.0$
+
+DM software is currently versioned using two digits separated by a "." .
+For example Science Pipelines latest release:\\
+$17.0$
+The first number is the major version and the the second number is the minor version.
+
+
+\subsubsection{Branch Naming} \label{sec:branchnaming}
+
+In order to clearly distinguish release branches from tags and other branches, the following naming scheme is suggested:
+
+\begin{verbatim}
+b.<M>.<n>.x
+\end{verbatim}
+
+where \textbf{M} is the major version and \textbf{n} is the minor version starting from 0.
+
+The letter  \textbf{b} indicates that this is a branch.
+The letter \textbf{x} is a placeholder. It indicates that versions in a specific branch will have a fixed \textbf{M.n} but different patch level, where \textbf{0} means no patch.
+
+
+\newpage
+\section{Change control} \label{sec:changecontrol}
+
+The DMCCB is in charge of approving the changes which go in  all releases (major, minor, or patch).
+The DM Release Plan LDM-564 provides the expected schedule for major features, based on the P6 milestones \footnote{The DM Release Plan is in the process to be updated, see \jira{DM-17001}}.
+
+Changes to the release plan  thus are  at least partially controlled the LSST change control  process as  milestones are moved or impacted by other changing milestones. Other changes not covered by the milestones such as features or content need to be proposed to the DMCCB using RFC Jira issues.
+
+Patch release requests should use an RFC Jira issue which must be approved by the DMCCB.
+The issue should specify what needs to be fixed in the release and why, it shall specify DM issues that are requested to be included in the patch.
+
+This process is documented in \citeds{LDM-294}, sections 3.6 and 7.4.
+
+
+\subsection{Issue Management} \label{sec:issues}
+
+The release is identified in Jira using a release issue.
+
+All issues to be included in a release shall be added as blocking to the release issue.
+
+Note that by introducing the field \textit{Fix in Version(s)}, will permit having the same information in a much simpler way.
+Both release issue and \textit{Fix in Version(s)} fields can be used in parallel.
```

Comment:
"Fix in Version(s)" is a native Jira field, and it is not just about which issues block a release, but characterize an issue in terms of in which release it has been fixed.

Note that not all issues included in a release are marked as blocking for that release. Many issues may have been fixed into master before the release process starts and there will be no automatic way from Jira to know in which release they have been fixed.

Note also that a release issue will be probably needed also to trace the activity. 

Finally, developers should not be bothered about the introduction of this field, since in the vast majority of the cases it should be filled a posteriori. We should not change the workflow, at least not for this reason.

Reply:
Git already tells you which issues have been merged into a particular release. This field adds no value given our workflow; it's only more complexity for no reason.

Reply:
We can drop the point if this makes confusion since it is not on the critical path for improving the release process.

However, from a management point of view, this field is important.
All relevant information relevant for an issue shall be in the issue itself, without the need to go to the repository to see in which version it has been fixed.
This shall not represent any extra effort for the developer and its update shall be automatic.